java/src/main/c++/vowpalWabbit_learner_VWLearners.cc

```diff
@@ -26,6 +26,20 @@ JNIEXPORT void JNICALL Java_vowpalWabbit_learner_VWLearners_closeInstance(JNIEnv
   }
 }
 
+JNIEXPORT void JNICALL Java_vowpalWabbit_learner_VWLearners_saveModel(JNIEnv *env, jclass obj, jlong vwPtr, jstring filename)
+{ try
+  {
+    const char* utf_string = env->GetStringUTFChars(filename, NULL);
+    std::string filenameCpp(utf_string);
+    VW::save_predictor(*(vw*)vwPtr, filenameCpp);
+    env->ReleaseStringUTFChars(filename, utf_string);
+    env->DeleteLocalRef(filename);
```

Comment:
C++ doesn't have finally blocks, unfortunately. I believe the `std::string` copies the characters inside of itself, so we should be able to do the memory releases after it is constructed and before `save_predictor` is called. This will prevent leaks from errors while saving. The issue with releasing inside of the exception-converting try block is done throughout the library. I took the pattern used here from `read_example` in jni_base_learner.cc, which isn't doing anything to prevent leaks.